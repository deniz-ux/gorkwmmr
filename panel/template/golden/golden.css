body {
  display: flex;
  height: 100vh;
  font-family: roboto;
  margin: 0px;
}

.main-area {
  flex: auto;
  overflow: auto;
  position: relative;
}

.main-content {
  height: 100%;
}

#app-bar {
  position: absolute;
  height: 64px;
  z-index: 7;
  background-color: black;
  width: 100%;
  color: white;
  display: flex;
  align-items: center;
}

.header-adjust {
  padding-top: 64px;
}

.header-title {
  font-size: 1.5em;
  padding-left: 1.5em;
}

p.bk.golden-card-button {
  display: none;
}
