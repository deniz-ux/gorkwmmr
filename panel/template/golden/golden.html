{% extends base %}

<!-- goes in body -->
{% block postamble %}
<script type="text/javascript" src="http://code.jquery.com/jquery-1.11.1.min.js"></script>
<script type="text/javascript" src="https://golden-layout.com/files/latest/js/goldenlayout.js"></script>
<link type="text/css" rel="stylesheet" href="https://golden-layout.com/files/latest/css/goldenlayout-base.css" />
<link type="text/css" rel="stylesheet" href="//golden-layout.com/assets/css/goldenlayout-dark-theme.css" />
{% endblock %}

<!-- goes in body -->
{% block contents %}
<header class="app-bar" id="header">
  <div class="">
    <section class="">
      <span class="header-title">{{ app_title }}</span>
	  {% for doc in docs %}
      {% for root in doc.roots %}
      {% if "header" in root.tags %}
      {{ embed(root) | indent(8) }}
      {% endif %}
      {% endfor %}
      {% endfor %}
    </section>
  </div>
</header>

{% if nav %}
<aside class="header-adjust" id="sidebar">
  <div class="nav-content">
    <div class="nav-list">
	  {% for doc in docs %}
	  {% for root in doc.roots %}
	  {% if "nav" in root.tags %}
      {{ embed(root) | indent(8) }}
	  {% endif %}
	  {% endfor %}
	  {% endfor %}
    </div>
  </div>
</aside>
{% endif %}

<div class="main-area header-adjust" id="main">
  <main class="main-content" id="main-content"></main>
</div>

<script>
  var config = {
    content: [{type: 'stack', content: [
    {% for doc in docs %}
    {% for root in doc.roots %}
    {% if "main" in root.tags %}
      {'type': 'component', 'componentName': 'view', componentState: {model: '{{ embed(root) }}', title: '{{ root.name}}' }},
    {% endif %}
    {% endfor %}
    {% endfor %}
    ]}]
  };

  var myLayout = new GoldenLayout(config, $('#main-content'));
  myLayout.registerComponent('view', function( container, componentState ){
    const {width, css_classes} = componentState
    if(width)
      container.on('open', () => container.setSize(width, container.height))
    if (css_classes)
      css_classes.map((item) => container.getElement().addClass(item))
    container.setTitle(componentState.title)
    container.getElement().html(componentState.model);
    container.on('resize', () => window.dispatchEvent(new Event('resize')))
    window.dispatchEvent(new Event('resize'));
  });
  myLayout.init()
</script>


{% endblock %}
